modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/portal/ApiWidgetHandler.java

```diff
@@ -0,0 +1,121 @@
+package com.enonic.xp.admin.impl.portal;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.enonic.xp.admin.widget.WidgetDescriptor;
+import com.enonic.xp.admin.widget.WidgetDescriptorService;
+import com.enonic.xp.api.ApiDescriptor;
+import com.enonic.xp.app.ApplicationKey;
+import com.enonic.xp.context.ContextAccessor;
+import com.enonic.xp.page.DescriptorKey;
+import com.enonic.xp.portal.PortalRequest;
+import com.enonic.xp.portal.controller.ControllerScriptFactory;
+import com.enonic.xp.resource.ResourceKey;
+import com.enonic.xp.security.PrincipalKeys;
+import com.enonic.xp.security.RoleKeys;
+import com.enonic.xp.web.WebException;
+import com.enonic.xp.web.WebRequest;
+import com.enonic.xp.web.WebResponse;
+import com.enonic.xp.web.api.ApiHandler;
+
+@Component(immediate = true, service = ApiHandler.class, property = ApiHandler.APPLICATION_KEY_PROPERTY + "=widget")
+public class ApiWidgetHandler
+    implements ApiHandler
+{
+    private static final Pattern WIDGET_API_PATTERN = Pattern.compile( "^/(_|api)/widget/(?<appKey>[^/]+)/(?<widgetKey>[^/]+)" );
```

Comment:
How about mapping it like this `/_/widget/<appKey>:<widgetKey>`?  Where `apiKey` is combined and in this case it fits  good to common approach with Slash API and Harmonized API.